test/chart/LineChart.spec.tsx

```diff
@@ -369,6 +616,126 @@ describe('<LineChart />', () => {
   });
 });
 
+describe('<LineChart /> and various data sources', () => {
+  const randData = generateMockData(10, 38405);
+  const data1 = randData.slice(0, 5);
+  const data2 = randData.slice(5);
+
+  function expectLabels(container: HTMLElement, expectedTextContents: ReadonlyArray<string>): void {
+    const labels = container.querySelectorAll('.recharts-label');
+    const actualTextContents = Array.from(labels).map(el => el.textContent);
+    expect(actualTextContents).toEqual(expectedTextContents);
+  }
+
+  it('should render chart with one line and data on root chart', () => {
+    const { container } = render(
+      <LineChart width={400} height={400} data={data1}>
+        <Line isAnimationActive={false} label dataKey="x" />
+        <XAxis dataKey="label" />
+      </LineChart>,
+    );
+    expectLabels(container, ['258', '295', '193', '168', '117']);
+    expectXAxisTicks(container, ['Iter: 0', 'Iter: 1', 'Iter: 2', 'Iter: 3', 'Iter: 4']);
+  });
+
+  it('should render chart with three lines and data on root chart', () => {
+    const { container } = render(
+      <LineChart width={400} height={400} data={data1}>
+        <Line isAnimationActive={false} label dataKey="x" />
+        <Line isAnimationActive={false} label dataKey="y" />
+        <XAxis dataKey="label" />
+      </LineChart>,
+    );
+    expectLabels(container, ['258', '295', '193', '168', '117', '597', '745', '657', '538', '762']);
+    expectXAxisTicks(container, ['Iter: 0', 'Iter: 1', 'Iter: 2', 'Iter: 3', 'Iter: 4']);
+  });
+
+  it('should render the chart when the same data are defined on Line elements and not on the chart', () => {
+    const { container } = render(
+      <LineChart width={400} height={400}>
+        <Line data={data1} isAnimationActive={false} label dataKey="x" />
+        <Line data={data1} isAnimationActive={false} label dataKey="y" />
+        <XAxis dataKey="label" />
+      </LineChart>,
+    );
+    expectLabels(container, ['258', '295', '193', '168', '117', '597', '745', '657', '538', '762']);
+    // sic! the XAxis renders all ticks twice!
```

Comment:
I think if you specify allowDuplicatedCategory={false} it won't render them twice

Can we add a case for that?